okhttp/src/main/java/okhttp3/Request.java

```diff
@@ -217,6 +228,11 @@ public Builder headers(Headers headers) {
       return this;
     }
 
+    public Builder cacheKey(String cacheKey) {
```

Comment:
There is a high bar to public API additions or changes.  This should be discussed and would deserve it's own PR separate from the listener change.

I think cacheKey semantics alone would warrant a fair amount of discussion to get right if it is likely to be accepted.